pkg/reconciler/deployment.go

```diff
@@ -39,29 +47,36 @@ type DeploymentReconciler struct {
 	KubeClientSet kubernetes.Interface
 }
 
-func (r *DeploymentReconciler) ReconcileDeployment(ctx context.Context, owner kmeta.OwnerRefable, expected *appsv1.Deployment) (*appsv1.Deployment, pkgreconciler.Event) {
+// ReconcileDeployment reconciles deployment resource for SampleSource
+func (r *DeploymentReconciler) ReconcileDeployment(
+	ctx context.Context,
+	owner kmeta.OwnerRefable,
+	binder *sourcesv1.SinkBinding,
+	expected *appsv1.Deployment,
+) (*appsv1.Deployment, *sourcesv1.SinkBinding, pkgreconciler.Event) {
+	syncSink(ctx, binder, expected.Spec.Template.Spec)
```

Comment:
would it make more sense to just call `binder.Do()` directly here without all the deepcopy'ing and equality checks if (if I understand it correctly) we're looking to just `Do()` the transformation?

Reply:
Makes sense. now only `podSpecSync` returns bool. Thanks!